lib/ipam/ipam_test.go

```diff
@@ -514,15 +461,53 @@ var _ = testutils.E2eDatastoreDescribe("IPAM tests", testutils.DatastoreEtcdV2,
 
 	DescribeTable("ReleaseIPs: requested IPs to be released vs actual unallocated IPs",
 		func(inIP net.IP, cleanEnv bool, pool []string, assignIP net.IP, autoAssignNumIPv4 int, expUnallocatedIPs []cnet.IP, expError error) {
-			unallocatedIPs, outError := testIPAMReleaseIPs(inIP, pool, cleanEnv, assignIP, autoAssignNumIPv4, config)
+			inIPs := []cnet.IP{cnet.IP{inIP}}
+
+			// If we cleaned the datastore then recreate the pools.
+			if cleanEnv {
+				bc.Clean()
+				deleteAllPools()
+			}
+			for _, v := range pool {
+				applyPool(v, true)
+			}
+
+			if len(assignIP) != 0 {
+				err := ic.AssignIP(AssignIPArgs{
+					IP: cnet.IP{assignIP},
+				})
+				if err != nil {
+					Fail(fmt.Sprintf("Error assigning IP %s", assignIP))
+				}
+
+				// Re-initialize it to an empty slice to flush out any IP if passed in by mistake.
+				inIPs = []cnet.IP{}
+
+				inIPs = append(inIPs, cnet.IP{assignIP})
+
+			}
+
+			if autoAssignNumIPv4 != 0 {
+				assignedIPv4, _, _ := ic.AutoAssign(AutoAssignArgs{
```

Comment:
why is the err ignored here?

Reply:
dunno - I just moved the code from a standalone function to inside here since it was only called from this one test.

I daresay we should assert the other things though, I'll add.